### PR TITLE
Add possible fix of error handling

### PR DESCRIPTION
**Pull request by andriisoldatenko**
fix https://github.com/astronomer/astro-cli/issues/182

### CODE REVIEW

File: houston/houston.go [hunk 1]
```diff
@@ -12,6 +12,22 @@ import (
 	"github.com/astronomer/astro-cli/pkg/httputil"
 )
 
+
+type ErrPermissions struct {
```
Comment (andriisoldatenko) [inline houston/houston.go]: for now skip this, i'll need more details how we plan to handle it.

File: houston/houston.go [hunk 2]
```diff
@@ -85,7 +95,16 @@ func (c *Client) Do(doOpts httputil.DoOptions) (*HoustonResponse, error) {
 
 	// Houston Specific Errors
 	if decode.Errors != nil {
-		return nil, errors.New(decode.Errors[0].Message)
+		err = errors.New(decode.Errors[0].Message)
+		if err != nil {
+			switch err.(type) {
+			case *JWTTokenError:
```
Comment (andriisoldatenko) [inline houston/houston.go]: @schnie 
I've added possibility to extend to catch different error types.

File: houston/houston.go [hunk 3]
```diff
@@ -12,6 +12,16 @@ import (
 	"github.com/astronomer/astro-cli/pkg/httputil"
 )
 
+
+// JWTTokenError records an error and the operation and file path that caused it.
+type JWTTokenError struct {
```
Comment (andriisoldatenko) [inline houston/houston.go]: we can move all errors to `errors.go`

**Comment by andriisoldatenko** [comment]
```
astro-cli git:(more-descriptive-error-message-when-token-expires) ✗ go run main.go deployment list
Error: Your token has expired. Please log in again.
exit status 1
```

**Comment by andriisoldatenko** [comment]
```
go run main.go deployment list
 NAME     DEPLOYMENT NAME     ASTRO     DEPLOYMENT ID
```

**Comment by schnie** [comment]
LGTM!

**Pull request merged**